pythonProject/theory questions/exam.txt

```diff
@@ -0,0 +1,72 @@
+Questions:
+2-marks:
+1) What is Selenium?
+selenium is a web application automation framework.
+it has more tools and libraries to automate application
+
+2) Command to install selenium.
+pip install selenium
+3) What are the locators in Selenium?
+ID,name,classname,xpath,linked text,partially linked text,css selector
+4) What is the difference between find_element_by_* and find_elements_by_* in Selenium?
+find _element_by-*..... is used to find out the only single element by the locator
+ but ,find_element_by_* is used to find out the multiple element
+5) How do you manage browser windows in Selenium?
+driver.WindowHandles()
+driver.currentWindowHandle()
+switch_to()
+switch_to.(driver.window_handles(index))
+
+6) What are the frameworks used in selenium. Explain
+pytest-pytest specially  has @pytest.fixture using this we reduce the duplication of code
+ ------here we have @pytest.mark.order()using this we can customize our test funcs
+ @pytest.mark.parameterization()---multiple run test---used for data driven test so we pass the different data
+ depending on that test will run multiple tie
+unittest-mostly used by development team
+testproject-
+robot-key driven development framework....
+maintainability,clarity,scalability,reporting these are the mager advantage in this
+
+behave(bdd)-collaboration is the best thing here..
+because of this we can collaborate with development team,QA and
+**Business partner(non technical person)...
+maintainability,clarity,simple language,gherkin keyword,collaboration are main advantage
+7) What is selenium Grid and how does it work?
+selenium grid is used for  handle different browser in same time...
+8) How do you initialize web driver in selenium?
+By using service object
+driver=webdriver.Chrome(service=Service(ChromeWebdriverManger().install())
+9) How do you find a element in selenium?
```

Comment:
Why didnt we share the code snippet? 

driver.find_element(By.Xpath, 'locator'). 

0 marks.